src/event/event.rs

```diff
@@ -78,6 +78,13 @@ impl Default for Event {
 }
 
 impl Event {
+    /// Returns an `Iterator` for `Attributes`
+    pub fn attributes_iter(&self) -> impl Iterator<Item = (&str, AttributeValue<'_>)> {
```

Comment:
you need to explicit the lifetime here i think:

```rust
 pub fn <'a> attributes_iter(&'a self) -> impl Iterator<Item = (&'a str, AttributeValue<'a>)>
```

Reply:
I think its fine because the anonymous lifetime here takes care of that, unless you know a particular reason or case where this would fail

Reply:
Does it shows a warning that tells you that you can remove the lifetime definition?

Reply:
nope

Reply:
so that means that you need it, because otherwise the compiler could infer the wrong one. If you don't need it, then the compiler tells you "the one you wrote is the same i inferred"

Reply:
done!